File: src/main/java/cs/java/collections/CSMapImpl.java
package cs.java.collections;

public class CSMapImpl<K, V> extends java.util.HashMap<K, V> implements CSMap<K, V> {

    public boolean hasKey(K key) {
        return super.containsKey(key);
    }

    public boolean hasValue(V value) {
        return super.containsValue(value);
    }

    public V value(K key) {
        return super.get(key);
    }

}

File: library/base/src/main/java/renetik/android/java/collections/CSMapImpl.kt
package renetik.android.java.collections

class CSMapImpl<K, V> : java.util.HashMap<K, V>, CSMap<K, V> {

    constructor() {}

    constructor(map: Map<out K, V>) : super(map) {}

    override fun hasKey(key: K) = super.containsKey(key)

    override fun hasValue(value: V) = super.containsValue(value)

    override fun value(key: K) = super.get(key)

}
